lms/djangoapps/courseware/management/commands/tests/test_dump_to_neo4j.py

```diff
@@ -0,0 +1,154 @@
+        self.assertEqual(mock_transaction.create.call_count, 18)
+        self.assertEqual(mock_transaction.run.call_count, 2)
+
+    @mock.patch('courseware.management.commands.dump_to_neo4j.Graph')
+    def test_dump_to_neo4j_rollback(self, mock_graph_class):
+        """
+        Tests that the management command handles the case where there's
+        an exception trying to write to the neo4j database.
+        """
+        mock_graph = mock_graph_class.return_value
+        mock_transaction = mock.Mock()
+        mock_graph.begin.return_value = mock_transaction
+        mock_transaction.run.side_effect = ValueError('Something went wrong!')
+
+        call_command('dump_to_neo4j')
+
+        self.assertEqual(mock_graph.begin.call_count, 2)
+        self.assertEqual(mock_transaction.commit.call_count, 0)
+        self.assertEqual(mock_transaction.rollback.call_count, 2)
+
+    @mock.patch('django.conf.settings.NEO4J_CONFIG', None)
+    def test_dump_to_neo4j_no_config(self):
+        """
+        Tests that the command errors out if there isn't a configuration
+        file found
+        """
+        with self.assertRaises(CommandError):
+            call_command('dump_to_neo4j')
+
+
+@ddt.ddt
+class TestModuleStoreSerializer(TestDumpToNeo4jCommandBase):
+    """
+    Tests for the ModuleStoreSerializer
+    """
+    def setUp(self):
+        super(TestModuleStoreSerializer, self).setUp()
+        self.modulestore_serializer = ModuleStoreSerializer()
+
+    def test_serialize_item(self):
+        """
+        Tests the serialize_item method.
+        """
+        fields, label = self.modulestore_serializer.serialize_item(self.course)
+        self.assertEqual(label, "course")
+        self.assertIn("edited_on", fields.keys())
+        self.assertIn("display_name", fields.keys())
+        self.assertIn("org", fields.keys())
+        self.assertIn("course", fields.keys())
+        self.assertIn("run", fields.keys())
+        self.assertIn("course_key", fields.keys())
+        self.assertNotIn("checklist", fields.keys())
+
+    def test_serialize_course(self):
+        """
+        Tests the serialize_course method.
+        """
+        nodes, relationships = self.modulestore_serializer.serialize_course(
+            self.course.id
+        )
+        self.assertEqual(len(nodes), 9)
+        self.assertEqual(len(relationships), 7)
+
+    @ddt.data(*ITERABLE_NEO4J_TYPES)
+    def test_coerce_types_iterable(self, iterable_type):
+        """
+        Tests the coerce_types helper method for iterable types
+        """
+        example_iterable = iterable_type([object, object, object])
+
+        # each element in the iterable is not unicode:
+        self.assertFalse(any(isinstance(tab, unicode) for tab in example_iterable))
+        # but after they are coerced, they are:
+        coerced = self.modulestore_serializer.coerce_types(example_iterable)
+        self.assertTrue(all(isinstance(tab, unicode) for tab in coerced))
```

Comment:
The use of `unicode` here and just above should be replaced with six.text_type to work under Python 3